packages/core/database/src/utils/identifiers/index.ts

```diff
@@ -21,25 +41,164 @@ type NameOptions = {
   maxLength?: number;
 };
 
-export const getNameFromTokens = (tokens: NameToken[], max: number = MAX_DB_IDENTIFIER_LENGTH) => {
-  const fullLength = tokens
+// returns a hash of length len
+export function createHash(data: string, len: number): string {
+  if (!_.isInteger(len) || len <= 0) {
+    throw new Error(`createHash length must be a positive integer, received ${len}`);
+  }
+
+  // TODO: shake256 is based on SHA-3 and is slow, we don't care about cryptographically secure, only uniqueness and speed
+  //       investigate alternatives before releasing this. But it is only run on startup, so it should be fine.
+  const hash = crypto.createHash('shake256', { outputLength: Math.ceil(len / 2) }).update(data);
+  return hash.digest('hex').substring(0, len);
+}
+
+export function tokenWithHash(name: string, len: number) {
```

Comment:
Can we write some JS doc where this is used my assumption is something like this:

```js
/**
* @internal
* @description takes a string of a DB table and hashes it down to a shorter length to not break the constraints of DBs. This is typically done only when a schema is named or renamed.
*/
```